src/System.IO.FileSystem/src/System/IO/Win32FileSystemEnumerable.cs

```diff
@@ -398,10 +405,10 @@ private bool IsResultIncluded(ref Interop.Kernel32.WIN32_FIND_DATA findData, out
         }
 
         [SecurityCritical]
-        private void HandleError(int errorCode, string path)
+        private Exception HandleError(int errorCode, string path)
```

Comment:
I'm conflicted- it does more than just get an exception. It gets a bit wordy so I didn't end up changing it.